programs/perpetuals/src/state/custody.rs

```diff
@@ -181,6 +211,62 @@ impl Custody {
             && self.oracle.validate()
             && self.pricing.validate()
             && self.fees.validate()
+            && self.borrow_rate.validate()
+    }
+
+    pub fn update_borrow_rate(&mut self, curtime: i64) -> Result<()> {
```

Comment:
it is really smart to run this function every time a user interacts with this custody.

I am just thinking in case a pool didn't have any interaction for a long period of time, those positions will be at risk because once the borrow rates add up, the position might get liquidated straight away. Is it better to add a permissionless function for anyone to update the borrow rate?

Reply:
Are you thinking about the situation where a taker has a an ever increasing profit thus leading to a an increased `assets.locked`? In that case the real borrow rate will deviate from the borrow rate stored in state. 

This would present a wrong interest when the users tries to get the PnL.  

Can't the liquidator just handle it?

Reply:
@hjchan good point, get_liquidation_price and get_liquidation_state instructions don't update borrow rates, so in the scenario with zero interactions with the protocol, user's position might appear healthy while it is not. I'll update the code.

Reply:
So liquidators would mainly be used as cranks?

Reply:
I've updated borrow rate handling, most recent value is calculated before a position is about to open/close/liquidated as well as for getters like get_liquidation_price. No need for a separate crank for this. Currently only liquidator bot requires a crank, but it is permissionless, so anyone can run it.